convert2rhel/unit_tests/systeminfo_test.py

```diff
@@ -346,19 +259,26 @@ def test_corresponds_to_rhel_eus_release_eus_override(major, minor, expected, mo
 
 
 @pytest.mark.parametrize(
-    ("system_release_content", "expected"),
+    ("system_release_content", "key", "value"),
     (
-        ("CentOS Linux release 8.1.1911 (Core)", "Core"),
-        ("Oracle Linux Server release 7.8", None),
+        ("CentOS Linux release 8.1.1911 (Core)", "distribution_id", "Core"),
+        ("Oracle Linux Server release 7.8", "name", "Oracle Linux Server"),
+        ("Oracle Linux Server release 7.8", "distribution_id", None),
+        ("CentOS Stream release 8", "id", "centos"),
+        ("CentOS Linux release 8.1.1911 (Core)", "version", Version(8, 1)),
+        ("CentOS Stream release 8", "version", Version(8, 10)),
+        ("Red Hat Enterprise Linux release 8.10 Beta (Ootpa)", "version", Version(8, 10)),
     ),
 )
-def test_get_system_distribution_id(system_release_content, expected):
-    assert system_info._get_system_distribution_id(system_release_content) == expected
+def test_parse_system_release_content_from_string(system_release_content, key, value):
```

Comment:
I think we should, at least, have more parameters here to validate the bad scenario for the other parts, for example, what happens if there is no `version` and `name` in the content string?

Reply:
It is little likely that this scenario would happen on a real system, and if that happens, it is probably better to just tell the user that their system is beyond broken, but still, I think that it would be good to catch that scenario, at least so if we identify we can give the user a good response like "hey, your system is beyond broken because it doesn't have X,Y or Z. Please fix it"

Reply:
I have added a separate `test_fail_to_parse_system_release_content_from_string()` test below for a failure scenario. Is it what you are referring to? Or you expect something else?

We are using a single regular expression to match the entire string, so I can not easily tell the user which part of the string is missing. I can say no matches found. And that's what I do in that test.

Reply:
Oh, now I see. I think I have missed that test (or you introduced it recently), but that sounds good to me. I will resolve this comment.